platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java

```diff
@@ -195,25 +194,33 @@ public class ConsensusImpl extends ThreadSafeConsensusInfo implements Consensus
     /**
      * Constructs an empty object (no events) to keep track of elections and calculate consensus.
      *
-     * @param config consensus configuration
+     * @param platformContext  the platform context containing configuration
      * @param consensusMetrics metrics related to consensus
-     * @param addressBook the global address book, which never changes
-     * @param ancientMode describes how we are currently computing "ancientness" of events
+     * @param addressBook      the global address book, which never changes
      */
     public ConsensusImpl(
-            @NonNull final ConsensusConfig config,
+            @NonNull final PlatformContext platformContext,
             @NonNull final ConsensusMetrics consensusMetrics,
-            @NonNull final AddressBook addressBook,
-            @NonNull final AncientMode ancientMode) {
-        super(config, new SequentialRingBuffer<>(ConsensusConstants.ROUND_FIRST, config.roundsExpired() * 2));
-        this.config = config;
+            @NonNull final AddressBook addressBook) {
+        super(
+                platformContext.getConfiguration().getConfigData(ConsensusConfig.class),
+                new SequentialRingBuffer<>(
+                        ConsensusConstants.ROUND_FIRST,
+                        platformContext
+                                        .getConfiguration()
+                                        .getConfigData(ConsensusConfig.class)
+                                        .roundsExpired()
+                                * 2));
```

Comment:
I'd suggest that we make the parent class `ThreadSafeConsensusInfo` also take a platform context as an argument. Also, it's probably better to move the construction of the `SequentialRingBuffer` into the parent class as well.